### PR TITLE
added scope notebook

### PR DESCRIPTION
**Pull request by jasonmsetiadi**
Please comment for edit suggestions

### CODE REVIEW

**Comment by mcoughlin** [comment]
@jasonmsetiadi Ugh. Yeah can we do this as a python script instead?

**Comment by jasonmsetiadi** [comment]
i just added the scope_jason.py file can you see it?

File: scope_jason.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+disagreeing_scores = scores.iloc[disagreements,:]
+
+stats = get_stats(disagreeing_scores['_id'])
+data = pd.merge(disagreeing_scores,stats,left_on='_id',right_on='_id')
+data['train'] = np.isin(data['_id'],trainingset['ztf_id'])
+sample = data[~data['train']]
+
+def upload():
+    import requests
+    token = "9f78fcea-61f0-4b71-8b9f-d1b1b573fd4d" # Jason's Fritz token
```
Comment (mcoughlin) [inline scope_jason.py:71]: This should be in a file imported.

File: scope_jason.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    headers={"Authorization": f"token {token}"}
+
+    for index, row in sample.iterrows():
+        #print(row)
+        i = row['_id']
+        period = row['period']
+        
+        # upload 
+        json = { "catalog": "ZTF_sources_20210401",
+                  "group_ids": [371]} #group id for the upload location
```
Comment (mcoughlin) [inline scope_jason.py:88]: The group_id should be an argument to a function.

File: scope_jason.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          headers=headers,)
+
+        # get objid
+        try:
+            obj_id = response.json()['data']['obj_id']
+        except:
+            print('failed to upload target')
+            continue
+
+        print(obj_id)
```
Comment (mcoughlin) [inline scope_jason.py:102]: errant print statement.

File: scope_jason.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
+import pandas as pd
+import pdb
+import json
+from penquins import Kowalski
+import numpy as np
+
+def get_highscoring_objects(otype='vnv',condition="$or",
```
Comment (mcoughlin) [inline scope_jason.py:7]: can we put these functions in scope and import them from there?

File: scope_jason.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+                 "projection": {}
+             },
+             "kwargs": {     }
+             }
+
+    r = G.query(q)
+
+    return pd.DataFrame(r['data'])
+
+def get_stats(ids):
```
Comment (mcoughlin) [inline scope_jason.py:31]: same as above.

File: scope_jason.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+             "kwargs": {     }
+             }
+        for i in ids
+    ]
+    rs = G.batch_query(qs, n_treads=32)
+
+    return pd.DataFrame([s['data'][0] for s in rs])
+
+# load dataset
+trainingset = pd.read_csv('dataset.d15.csv')
```
Comment (mcoughlin) [inline scope_jason.py:49]: Make this file path a command line argument.

**Review by mcoughlin — CHANGES_REQUESTED** [review]
First review. Retag me when ready.

File: scope_jason.py (hunk 3 above)
Comment (jasonmsetiadi) [reply scope_jason.py:102]: what do you mean by errant?

File: scope_jason.py (hunk 6 above)
Comment (jasonmsetiadi) [reply scope_jason.py:49]: could you explain more how to do that?

Comment (mcoughlin) [reply scope_jason.py:49]: https://realpython.com/python-command-line-arguments/

File: scope_jason.py (hunk 3 above)
Comment (mcoughlin) [reply scope_jason.py:102]: You don't need a print statement in production. I suggest removing this unless it has more verbose information for the user.

Comment (jasonmsetiadi) [reply scope_jason.py:102]: got it michael!

**Comment by jasonmsetiadi** [comment]
@mcoughlin Hi Michael, I think I have made the changes you asked. Hopefully they work properly.

File: scope_jason.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    headers={"Authorization": f"token {token}"}
+
+    for index, row in sample.iterrows():
+        #print(row)
+        i = row['_id']
+        period = row['period']
+        
+        # upload 
+        json = { "catalog": "ZTF_sources_20210401",
+                  "group_ids": [group_id]}
```
Comment (mcoughlin) [inline scope_jason.py:58]: how is this specified?

File: scope_jason.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        # get objid
+        try:
+            obj_id = response.json()['data']['obj_id']
+        except:
+            continue
+
+        # annotate
+        url = 'https://fritz.science/api/sources/%s/annotations' %obj_id
+        json = { "origin": "ML_DR5_d15_v1",
+              "group_ids": [348]}
```
Comment (mcoughlin) [inline scope_jason.py:74]: Similar need to make this configurable.

**Review by mcoughlin — COMMENTED** [review]
Getting closer. As much as possible, think of how to generalize this script.

File: scope_jason.py (hunk 7 above)
Comment (jasonmsetiadi) [reply scope_jason.py:58]: just by upload(integer)

File: scope_jason.py (hunk 8 above)
Comment (jasonmsetiadi) [reply scope_jason.py:74]: you mean just change the 348 to group_id right?

Comment (mcoughlin) [reply scope_jason.py:74]: yes

**Comment by jasonmsetiadi** [comment]
@mcoughlin i have changed the id to CL argument

File: scope.py [hunk 9]
```diff
@@ -750,6 +750,46 @@ def test(self):
             # clean up after thyself
             (path_mock / dataset).unlink()
 
+    def get_highscoring_objects(otype='vnv',condition="$or",
```
Comment (mcoughlin) [inline scope.py:753]: Maybe this stuff belongs in its own file in /scope? like /scope/fritz.py or something?

Comment (mcoughlin) [reply scope.py:753]: @janvanroestel ?

**Review by mcoughlin — APPROVED** [review]
@janvanroestel should merge once he is happy.

**Comment by mcoughlin** [comment]
@jasonmsetiadi can you please run
pre-commit install
in your git directory and push the modified script once it passes?

**Comment by jasonmsetiadi** [comment]
I think i have done it Michael

**Comment by jasonmsetiadi** [comment]
@mcoughlin i think everything has passed now

**Review by janvanroestel — APPROVED** [review]

**Pull request merged**